components/geolocation/geolocation.service.ts

```diff
@@ -0,0 +1,338 @@
+import * as Gyro from '../../lib/gyronorm_updated_206';
+import FULLTILT from './fulltilt';
+import Feature from 'ol/Feature';
+import Geolocation from 'ol/Geolocation';
+import Rotate from 'ol/control/Rotate';
+import VectorLayer from 'ol/layer/Vector';
+import {Circle, Fill, Stroke, Style} from 'ol/style';
+import {Circle as CircleGeom, Point} from 'ol/geom';
+import {GyroNorm} from '../../lib/gyronorm_updated_206';
+import {HsLayoutService} from '../layout/layout.service';
+import {HsMapService} from '../map/map.service';
+import {HsUtilsService} from './../utils/utils.service';
+import {Injectable} from '@angular/core';
+import {Vector} from 'ol/source';
+import {toRadians} from 'ol/math';
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * @param HsMapService
+ * @param HsLayoutService
+ * @param HsUtilsService
+ */
+export class HsGeolocationService {
+  localization: boolean;
+  following: boolean;
+  gn: any;
+  positionFeature: Feature;
+  centering: boolean;
+  accuracyFeature: Feature;
+  stopCentering: any;
+  geolocation: any;
+  clicked: any;
+  cancelClick: boolean;
+  style: any;
+  position_layer: any;
+  constructor(
+    private HsMapService: HsMapService,
+    private HsLayoutService: HsLayoutService,
+    private HsUtilsService: HsUtilsService
+  ) {
+    /**
+     * @ngdoc property
+     * @name HsGeolocationService#localization
+     * @public
+     * @type {boolean} false
+     * @description Represents geolocalization state (on/off)
+     */
+    this.localization = false;
```

Comment:
You can set the initial value when declaring `localization: boolean = false; `. Constructor will be shorter that way.

Reply:
Cleaned up the code